tools/test-framework/src/types/topology.rs

```diff
@@ -0,0 +1,145 @@
+
+use ibc_relayer::chain::handle::ChainHandle;
+use ibc_relayer::foreign_client::ForeignClient;
+
+use crate::bootstrap::binary::chain::bootstrap_foreign_client;
+use crate::error::Error;
+
+pub enum TopologyTypes {
+    Linear,
+    Full,
+}
+
+impl FromStr for TopologyTypes {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "linear" => Ok(Self::Linear),
+            "full" => Ok(Self::Full),
+            _ => Err(Error::generic(eyre!("The topology `{s}` does not exist"))),
+        }
+    }
+}
+
+pub trait Topology<Handle: ChainHandle> {
+    fn get_topology(
+        &self,
+        chain_handles: &Vec<Handle>,
+    ) -> Result<HashMap<usize, HashMap<usize, ForeignClient<Handle, Handle>>>, Error>;
+}
+
+pub struct FullyConnectedTopology;
+
+impl<Handle: ChainHandle> Topology<Handle> for FullyConnectedTopology {
+    fn get_topology(
+        &self,
+        chain_handles: &Vec<Handle>,
+    ) -> Result<HashMap<usize, HashMap<usize, ForeignClient<Handle, Handle>>>, Error> {
+        let mut foreign_clients: HashMap<usize, HashMap<usize, ForeignClient<_, _>>> =
+            HashMap::new();
+
+        for (i, handle_a) in chain_handles.iter().enumerate() {
+            let mut foreign_clients_b = HashMap::new();
+
+            for (j, handle_b) in chain_handles.iter().enumerate() {
+                let foreign_client =
+                    bootstrap_foreign_client(handle_a, handle_b, Default::default())?;
+
+                foreign_clients_b.insert(j, foreign_client);
+            }
+
+            foreign_clients.insert(i, foreign_clients_b);
+        }
+        Ok(foreign_clients)
+    }
+}
+
+pub struct LinearTopology;
+
+impl<Handle: ChainHandle> Topology<Handle> for LinearTopology {
+    fn get_topology(
+        &self,
+        chain_handles: &Vec<Handle>,
+    ) -> Result<HashMap<usize, HashMap<usize, ForeignClient<Handle, Handle>>>, Error> {
+        let mut foreign_clients: HashMap<usize, HashMap<usize, ForeignClient<_, _>>> =
+            HashMap::new();
+
+        let last_index = chain_handles.len() - 1;
+        for (i, _) in chain_handles.iter().enumerate() {
+            let mut clients = HashMap::new();
+            if i < last_index {
+                let client = bootstrap_foreign_client(
+                    &chain_handles[i],
+                    &chain_handles[i + 1],
+                    Default::default(),
+                )?;
+                clients.insert(i + 1, client);
+            }
+            if i > 0 {
+                let client = bootstrap_foreign_client(
+                    &chain_handles[i],
+                    &chain_handles[i - 1],
+                    Default::default(),
+                )?;
+                clients.insert(i - 1, client);
+            }
+
+            foreign_clients.insert(i, clients);
+        }
+        Ok(foreign_clients)
+    }
+}
+
+pub fn get_topology<Handle: ChainHandle>(value: &str) -> Box<dyn Topology<Handle>> {
```

Comment:
How about:

```suggestion
pub fn bootstrap_topology<Handle: ChainHandle>(topology: TopologyType) -> Box<dyn Topology<Handle>> {
```

and leave the parsing of the topology string to the caller?

Reply:
done in b035a6c4f501d03ffc7ae74fe2192371538494d5